isisd/isisd.c

```diff
@@ -595,6 +589,68 @@ static int isis_vrf_delete(struct vrf *vrf)
 	return 0;
 }
 
+static void isis_set_redist_vrf_bitmaps(struct isis *isis, bool set)
```

Comment:
This is very similar to `isis_redist_update_zebra_subscriptions()`. Wouldn't it be possible to merge both functions together?

Overall these changes look good to me (important bug fixes), but it's a bit scary how complex the redistribute handling has gotten. Long term we should consider using VRF names instead of VRF IDs when sending redistribute requests to zebra, since VRF names are immutable. That way we wouldn't need to sync with zebra whenever a VRF is disabled, enabled, or just change its ID (for those adventurous enough to try it :)).

Reply:
I was thinking about merging them initially, but there are already a lot of indentation levels in `isis_redist_update_zebra_subscriptions`, and adding more conditions will make it horrible.

Let's keep them separate, and when we implement your suggestion about name registration, the second function will be simply removed.

Reply:
Sounds good to me :)